avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,107 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
 
-    kernel_version = platform.uname()[2]
+    def check_local_kernel_config(config_name):
+        """
+        Reports the configuration of $config_name of the current kernel
+
+        :param config_name: Name of kernel config to search
+        :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
+        :rtype: :class:`ModuleConfig`
+        """
+        kernel_version = platform.uname()[2]
+
+        config_file = '/boot/config-' + kernel_version
+        with open(config_file, 'r') as kernel_config:  # pylint: disable=W1514
+            for line in kernel_config:
+                line = line.split('=')
+
+                if len(line) != 2:
+                    continue
+                config = line[0].strip()
+                if config == config_name:
+                    option = line[1].strip()
+                    if option == "m":
+                        return ModuleConfig.MODULE
+                    else:
+                        return ModuleConfig.BUILTIN
+        return ModuleConfig.NOT_SET
+
+    def check_session_kernel_config(config_name, session):
+        """
+        Reports the configuration of $config_name of session's kernel
+
+        :param config_name: Name of kernel config to search
+        :param session: remote session
+        :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
+        :rtype: :class:`ModuleConfig`
+        """
+        kernel_version = session.cmd('uname -r').strip()
+        config_file = '/boot/config-' + kernel_version
+        config_info = session.cmd('grep ^"{}"= {}'.format(config_name,
+                                  config_file)).strip()
+
+        LOG.debug("Get config info {}".format(config_info))
+        line = config_info.split('=')
+        if len(line) != 2:
+            return ModuleConfig.NOT_SET
+
+        config = line[0].strip()
+        LOG.debug("Get config {}, target is {}".format(config, config_name))
```

Comment:
This needs to be:

```python
        LOG.debug("Get config %s, target is %s", config, config_name)
```

Or else pylint will complain about it (`logging-format-interpolation`).